app/controllers/admin/tenants_controller.rb

```diff
@@ -1,5 +1,58 @@
 class Admin::TenantsController < Admin::BaseController
   def index
-    @tenants = Tenant.select(:organization, :location, :slug).all
+    if !Tenant.unapproved.empty?
+      @unapproved = Tenant.unapproved.select(:organization,
+                                             :location,
+                                             :slug,
+                                             :id).all.order(:id)
+    else
+      @unapproved = nil
+    end
+
+    if !Tenant.inactive.empty?
+      @inactive = Tenant.inactive.select(:organization,
+                                         :location,
+                                         :slug,
+                                         :id).all.order(:id)
+    else
+      @inactive = nil
+    end
+
+    if !Tenant.actives.empty?
+      @actives = Tenant.actives.select(:organization,
+                                       :location,
+                                       :slug,
+                                       :id).all.order(:id)
+    else
+      @actives = nil
+    end
+  end
+
+  def edit
+    @tenant = Tenant.find(tenant_id)
+  end
+
+  def update
+    tenant = Tenant.find(tenant_id)
+    old_approved_status = tenant.approved
+    tenant.update(params_update)
+
+    if old_approved_status == false && tenant.approved == true
```

Comment:
There IS an edge case, where if someone changes it to unapproved and switches it back to approved it will resend emails.